src/core/IPosisProcess.d.ts

```diff
@@ -3,6 +3,8 @@ interface IPosisProcess {
     imageName: string; // image name (maps to constructor)
     id: PosisPID; // ID
     parentId: PosisPID; // Parent ID
-    log: IPosisLogger; // Logger 
+    log: IPosisLogger; // Logger
+    // For querying extension interfaces (instead of tying ourselves to "levels")
+    queryPosisInterface<T extends keyof PosisInterfaces>(interfaceId: T): PosisInterfaces[T] | undefined;
```

Comment:
Thats another small thing, my kernel doesn't currently touch or even know about queryPosisInterface, that function is handled by my ExtensionRegistry, tying it into my kernel would make a direct dependency on ExtensionRegistry, defeating the purpose of trying to seperate it to begin with. For an example, my current index.ts is this:
```typescript
import { BaseKernel } from "BaseKernel";
import { ProcessRegistry } from "ProcessRegistry";
import { ExtensionRegistry } from "ExtensionRegistry";

let pkernel = new BaseKernel();

ExtensionRegistry.register("baseKernel", pkernel);

export function loop() {
  pkernel.loop();
}
```
The kernel doesn't even import ExtensionRegistry or know it exists, with this it would have to in order to inject the function.